src/software/time/time.h

```diff
@@ -2,7 +2,7 @@
 
 #include <time.h>
 
-#include <cstdlib>
+#include <chrono>
```

Comment:
If we end up implementing the suggestions I made above, we should be able to remove this, thus "restoring" the abstraction here.